app/src/main/kotlin/com/wire/android/ui/home/conversations/CompositeMessagesViewModel.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.ui.home.conversations
+
+import androidx.annotation.VisibleForTesting
+import androidx.compose.runtime.mutableStateMapOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class CompositeMessagesViewModel @Inject constructor() : ViewModel() {
+
+    var pendingButtons = mutableStateMapOf<String, String>()
```

Comment:
also this is reaaaaaaaaly focused on the UI related behaviour I think, but since we introduced viewmodel scoped to the Composable component I think we can live with that, but I feel like we are loosin once again what the viewmodel are supposed to be doing 😢

Reply:
the idea here is to have this mutableStateMap to keep track of which buttons have been clicked.
the next PR will introduce type alias to make improve readability `Map<MessageId, ButtonID>`
this idea here is only one button per message is allowed to be clicked at a time.